Documentation for granting and revoking access for teams. (#44863)
grafana/grafana: docs/sources/http_api/access_control.md

[
    {
        "version": 4,
        "uid": "j08ZBi-nk",
        "name": "fixed:licensing:reader",
        "displayName": "Licensing reader",
        "description": "Read licensing information and licensing reports.",
        "group": "Licenses",
        "updated": "2022-02-03T14:19:50+01:00",
        "created": "0001-01-01T00:00:00Z",
        "global": false
    }
]
```

#### Status codes

| Code | Description                                                          |
| ---- | -------------------------------------------------------------------- |
| 200  | Set of assigned roles is returned.                                   |
| 403  | Access denied.                                                       |
| 500  | Unexpected error. Refer to body and/or server logs for more details. |

### Add a team role assignment

`POST /api/access-control/teams/:teamId/roles`

Assign a role to a specific team.

For bulk updates consider [Set team role assignments]({{< ref "#set-team-role-assignments" >}}).

#### Required permissions

`permission:delegate` scope ensures that users can only assign roles which have same, or a subset of permissions which the user has.

> For example, if a user does not have required permissions for creating users, they won't be able to assign a role to a team which will allow to do that. This is done to prevent escalation of privileges.

Reviewer: nit: I find this sentence slightly difficult to read. Maybe something like:
```suggestion
For example, if a user does not have the permissions required to create users, they won't be able to assign a role that contains these permissions. This is done to prevent escalation of privileges.
```

| Field Name | Data Type | Required | Description      |
| ---------- | --------- | -------- | ---------------- |
| roleUid    | string    | Yes      | UID of the role. |

#### Example response

```http
HTTP/1.1 200 OK
Content-Type: application/json; charset=UTF-8

{
    "message": "Role added to the team."
}
```

#### Status codes

| Code | Description                                                          |
| ---- | -------------------------------------------------------------------- |
| 200  | Role is assigned to a team.                                          |
| 403  | Access denied.                                                       |
| 404  | Role not found.                                                      |
| 500  | Unexpected error. Refer to body and/or server logs for more details. |

## Remove a user role assignment

`DELETE /api/access-control/teams/:teams/roles/:roleUID`

Revoke a role from a team.

For bulk updates consider [Set team role assignments]({{< ref "#set-team-role-assignments" >}}).

#### Required permissions

`permission:delegate` scope ensures that users can only unassign roles which have same, or a subset of permissions which the user has.

> For example, if a user does not have required permissions for creating users, they won't be able to unassign a role from a team which will allow to do that. This is done to prevent escalation of privileges.

Reviewer: nit: similarly as above, this is a bit difficult to parse.

> For example, if a user does not have required permissions for creating users, they won't be able to unassign a role from a team which will allow to do that. This is done to prevent escalation of privileges.

Author: ```suggestion
For example, if a user does not have the permissions required to create users, they won't be able to assign a role that contains these permissions. This is done to prevent escalation of privileges.```
